tests/test_diffeq/test_odefiltsmooth/test_odefiltsmooth.py

```diff
@@ -1,507 +1,108 @@
-
-    def test_filter_ivp_h_ioup1_ekf(self):
-        probsolve_ivp(self.ivp, step=self.step, which_prior="ioup1", method="eks1")
-
-    def test_filter_ivp_h_ioup2_ukf(self):
-        """UKF requires some evaluation-variance to have a positive definite innovation
-        matrix, apparently."""
-        probsolve_ivp(
-            self.ivp, step=self.step, evlvar=0.01, which_prior="ioup2", method="uks"
-        )
-
-    def test_filter_ivp_h_ioup3_kf(self):
-        probsolve_ivp(self.ivp, step=self.step, which_prior="ioup3", method="eks0")
-
-    def test_filter_ivp_mat32_kf(self):
-        probsolve_ivp(
-            self.ivp,
-            atol=self.tol,
-            rtol=self.tol,
-            which_prior="matern32",
-            method="eks0",
-        )
-
-    def test_filter_ivp_mat52_ekf(self):
-        probsolve_ivp(
-            self.ivp,
-            atol=self.tol,
-            rtol=self.tol,
-            which_prior="matern52",
-            method="eks1",
-        )
-
-    def test_filter_ivp_mat72_ukf(self):
-        """UKF requires some evaluation-variance to have a positive definite innovation
-        matrix, apparently."""
-        probsolve_ivp(
-            self.ivp,
-            atol=self.tol,
-            rtol=self.tol,
-            evlvar=0.01,
-            which_prior="matern72",
-            method="uks",
-        )
-
-    def test_filter_ivp_h_mat32_ekf(self):
-        probsolve_ivp(self.ivp, step=self.step, which_prior="matern32", method="eks1")
-
-    def test_filter_ivp_h_mat52_ukf(self):
-        """UKF requires some evaluation-variance to have a positive definite innovation
-        matrix, apparently."""
-        probsolve_ivp(
-            self.ivp, step=self.step, evlvar=0.01, which_prior="matern52", method="uks"
-        )
-
-    def test_filter_ivp_h_mat72_kf(self):
-        probsolve_ivp(self.ivp, step=self.step, which_prior="matern72", method="eks0")
-
-
-class TestPreconditioning(unittest.TestCase):
-    """Solver with high order and small stepsize should work up to a point where
-    step**order is below machine precision."""
-
-    def setUp(self):
-        initdist = Constant(20 * np.ones(2))
-        self.ivp = ode.lotkavolterra([0.0, 1e-4], initdist)
-        self.step = 1e-5
-        self.prior = "ibm3"
 
-    def test_small_step_feasible(self):
-        """With the 'old' preconditioner, this is impossible because step**(2*order + 1)
-        is too small.
 
-        With the 'new' preconditioner, the smallest value that appears
-        in the solver code is step**order
-        """
-        probsolve_ivp(self.ivp, step=self.step, which_prior=self.prior, method="eks0")
+@pytest.fixture
+def ivp():
+    initrv = Constant(20.0 * np.ones(2))
+    return ode.lotkavolterra([0.0, 0.25], initrv)
+
+
+@pytest.mark.parametrize("method", ["EK0", "EK1"])
+@pytest.mark.parametrize(
+    "which_prior",
+    [
+        "IBM1",
+        "IBM2",
+        "IBM3",
+        "IBM5",
+        "IBM8",
+        "IBM11",
```

Comment:
we can get up to 11 already, even with bad initialisation :D 

IOUP and Matern seem capped at 4, though. No idea why.

Reply:
(this test only checks whether probsolve_ivp returns something useful for atol=rtol=1e-1. This is not a tough test, but I am happy about order 11 anyway. )